src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -24,19 +24,21 @@
  */
 package org.spongepowered.api.plugin;
 
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.util.regex.Pattern;
 
 /**
  * An annotation used to describe and mark a Sponge plugin.
  */
-@Target(TYPE)
-@Retention(RUNTIME)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
 public @interface Plugin {
 
+    Pattern ID_PATTERN = Pattern.compile("[a-z0-9-_]+");
```

Comment:
#1083 was started from a few comments above on this line, and is essentially discussing the need for additional parts of the identifier (as you suggest: a qualified package) or ways to replace the plugin IDs entirely. The class names were just one suggestion, I personally suggested something like `@Plugin(group = "net.minecrell", id = "testplugin")` before as an alternative. Currently our recommendation implied by docs etc. is essentially using an unique form of the plugin name as plugin ID.

I'm not against allowing qualified plugin IDs, but there were people against this in #1083, so I'd personally like to finish discussing the need for additional identifiers of plugin IDs before making any changes, because without a proper recommendation for the best practice we would soon have half plugins with qualified plugin IDs, and the other half with non-qualified plugin IDs.